deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvStages.java

```diff
@@ -132,6 +132,8 @@ public void update(
 
         if (environBean.getStage_type() == EnvType.DEV) {
             environBean.setAllow_private_build(true);
+        } else {
```

Comment:
What if stage type isn't changed but still provided in the request?

Reply:
Like DEV - > DEV or would it be NON-DEV -> NON-DEV?

If it's the 2nd one I have updated it to only update for changes where it goes from DEV -> NON-DEV.

Reply:
NON-DEV -> NON-DEV